textx/model.py

```diff
@@ -450,6 +467,32 @@ def _inner_resolve_link_rule_ref(cls, obj_name):
             result = _inner_resolve_link_rule_ref(obj_ref.cls,
                                                   obj_ref.obj_name)
             if result:
+                class RefRulePosition(object):
```

Comment:
This class should be moved out of this place. No need to create it multiple times, and the readability suffers.